plugins/cache-ehcache/src/main/java/org/opensearch/cache/store/disk/EhcacheDiskCache.java

```diff
@@ -162,7 +165,12 @@ private EhcacheDiskCache(Builder<K, V> builder) {
         this.ehCacheEventListener = new EhCacheEventListener(builder.getRemovalListener(), builder.getWeigher());
         this.cache = buildCache(Duration.ofMillis(expireAfterAccess.getMillis()), builder);
         List<String> dimensionNames = Objects.requireNonNull(builder.dimensionNames, "Dimension names can't be null");
-        this.cacheStatsHolder = new CacheStatsHolder(dimensionNames);
+
+        if (FeatureFlags.PLUGGABLE_CACHE_SETTING.get(builder.getSettings())) {
+            this.cacheStatsHolder = new CacheStatsHolder(dimensionNames);
+        } else {
+            this.cacheStatsHolder = new DummyCacheStatsHolder(dimensionNames);
```

Comment:
Lets rename this to NoopCacheStatsHolder. As dummy naming is more suitable in testing environment but noop is more production appropriate term,